src/_blog/metrics-in-go-libp2p.md

```diff
@@ -0,0 +1,182 @@
+---
+tags:
+- metrics
+- prometheus
+title: Metrics in go-libp2p
+description:
+date: 2023-06-15
+permalink: "/2023-06-15-metrics-in-go-libp2p/"
+author: Sukun Tarachandani
+---
+
+# Metrics in go-libp2p
+
+## Introduction
+
+libp2p is the core networking component for many projects such as IPFS, Filecoin, the Ethereum Beacon Chain, and more.
+We as maintainers of go-libp2p, want to be able to observe the state of libp2p components and also enable our users to do the same in their production systems.
+To that effect, we've been added instrumentation to collect metrics from various components over the last few months.
+In fact, they've already helped us debug some nuanced go-libp2p issues and helped with the development of features (discussed in detail below).
+Today, we'd like to share some of the choices we made, our learnings, and point you to resources that will help you monitor your deployments of go-libp2p.
+
+## Why Prometheus?
+
+We were first faced with the question of choosing a metrics collection and monitoring system. Among our choices were Prometheus, OpenCensus, and OpenTelemetry. The details of the discussion can be found [here](https://github.com/libp2p/go-libp2p/issues/1356).
+
+We noticed that [OpenCensus creates a lot of allocations](https://github.com/libp2p/go-libp2p/issues/1955), which would lead to increased GC pressure. OpenTelemetry's metrics API is still unstable as of writing this blog. In contrast, Prometheus is performant (zero-alloc) and ubiquitous. This allows us to add metrics without worrying too much about performance, even for frequently exercised code paths. 
+We also added ready-to-use Grafana dashboards, since we knew that Grafana is the preferred visualization tool of a lot of our users.
+
+## How Users can enable Metrics
+
+Metrics are enabled by default from go-libp2p [v0.26.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.26.0). All you need to do is setup a Prometheus exporter for the collected metrics. 
```

Comment:
```suggestion
Metrics are enabled by default since go-libp2p [v0.26.0](https://github.com/libp2p/go-libp2p/releases/tag/v0.26.0). All you need to do is setup a Prometheus exporter for the collected metrics. 
```